server/channels/app/platform/web_conn.go

```diff
@@ -254,9 +245,18 @@ func (ps *PlatformService) NewWebConn(cfg *WebConnConfig, suite SuiteIFace, runn
 		remoteAddress:      cfg.RemoteAddress,
 		xForwardedFor:      cfg.XForwardedFor,
 	}
-	wc.Active.Store(cfg.Active)
 
 	wc.SetSession(&cfg.Session)
+	userID := cfg.Session.UserId
+	if userID != "" {
+		ps.Go(func() {
```

Comment:
@agnivade One day, I'd love for us to document (i.e., through comments) why goroutines are needed in the first place. I suspect most of the logic in here could safely run outside, except maybe for something blocking the hub.

Reply:
Some of my thoughts are here: https://developers.mattermost.com/contribute/more-info/server/style-guide/#default-to-sync-instead-of-async :)

Reply:
> I suspect most of the logic in here could safely run outside, except maybe for something blocking the hub.

Yes, I did take a look. This is the issue.

Reply:
> > I suspect most of the logic in here could safely run outside, except maybe for something blocking the hub.
> 
> Yes, I did take a look. This is the issue.

Could you please elaborate a bit? The code here in question gets called whenever a websocket connection gets opened. Would it be a huge performance concern to wait for `SetStatusOnline` and `UpdateLastActivityAtIfNeeded` to complete?

Reply:
The issue is `ps.UpdateWebConnUserActivity(session, now)` sends a message into Hub. Which itself could block if hub is busy. That's why we cannot make this sync.

Reply:
Ah, thanks for the explanation. :+1: Added a comment via 2a42fb8d7cc4c1678c10a8472ff3a2f98b11c0ac.